api/src/main/java/com/epam/pipeline/manager/region/AzureRegionHelper.java

```diff
@@ -74,4 +96,127 @@ public AzureRegion mergeRegions(final AzureRegion originalRegion, final AzureReg
+        try {
+            final SharedKeyCredentials credentials = new SharedKeyCredentials(storageAccountName, storageAccountKey);
+            final RequestRetryOptions requestRetryOptions = new RequestRetryOptions(RetryPolicyType.EXPONENTIAL,
+                    MAX_TRIES_COUNT, TRY_TIMEOUT, null, null, null);
+            final PipelineOptions pipelineOptions = new PipelineOptions().withRequestRetryOptions(requestRetryOptions);
+            final ServiceURL serviceURL = new ServiceURL(new URL(
+                    String.format(BLOB_URL_FORMAT, storageAccountName)),
+                    StorageURL.createPipeline(credentials, pipelineOptions));
+            serviceURL.getProperties().blockingGet();
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    messageHelper.getMessage(MessageConstants.ERROR_AZURE_STORAGE_CREDENTIAL_INVALID), e);
+        }
+    }
+
+    void validateStoragePolicy(final AzurePolicy policy) {
+        if (Objects.isNull(policy)) {
+            return;
+        }
+
+        Assert.isTrue(StringUtils.isNotBlank(policy.getIpMax()) || StringUtils.isNotBlank(policy.getIpMin()),
+                messageHelper.getMessage(MessageConstants.ERROR_AZURE_IP_RANGE_IS_INVALID, policy.getIpMax(),
+                        policy.getIpMin()));
+        Assert.isTrue(isValidIpAddress(policy.getIpMax()),
+                messageHelper.getMessage(MessageConstants.ERROR_AZURE_IP_IS_INVALID, policy.getIpMax()));
+        Assert.isTrue(isValidIpAddress(policy.getIpMin()),
+                messageHelper.getMessage(MessageConstants.ERROR_AZURE_IP_IS_INVALID, policy.getIpMin()));
+        Assert.isTrue(isIpRangeValid(policy), messageHelper
+                .getMessage(MessageConstants.ERROR_AZURE_IP_RANGE_IS_INVALID, policy.getIpMax(), policy.getIpMin()));
+    }
+
+    private boolean isValidIpAddress(final String ip) {
+        return StringUtils.isBlank(ip) || InetAddressValidator.getInstance().isValid(ip);
+    }
+
+    void checkResourceGroupExistence(final String resourceGroup, final String authFilePath) {
+        Assert.isTrue(StringUtils.isNotBlank(resourceGroup), messageHelper.getMessage(
+                MessageConstants.ERROR_AZURE_RESOURCE_GROUP_NOT_FOUND, resourceGroup));
+        try {
+            final Azure client = buildClient(authFilePath);
+            Assert.isTrue(client.resourceGroups().contain(resourceGroup), messageHelper.getMessage(
+                    MessageConstants.ERROR_AZURE_RESOURCE_GROUP_NOT_FOUND, resourceGroup));
+        } catch (AuthenticationException e) {
+            throw new IllegalArgumentException(messageHelper.getMessage(
+                    MessageConstants.ERROR_AZURE_AUTH_FILE_IS_INVALID), e);
+        }
+    }
+
+    private Azure buildClient(final String authFile) {
+        try {
+            final Azure.Configurable builder = Azure.configure()
```

Comment:
I think this code is duplicated, could you please extract it to some common class.